nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3BinaryManager.java

```diff
@@ -428,6 +448,58 @@ protected FileStorage getFileStorage() {
         return new S3FileStorage();
     }
 
+    @Override
+    public String writeBlob(Blob blob) throws IOException {
+        // Attempt to do S3 Copy if the Source Blob provider is also S3
+        if (ManagedBlob.class.isAssignableFrom(blob.getClass())) {
```

Comment:
Why not ? ```blob instanceof ManagedBlob```

Reply:
`instanceof` only checks if it's an instance of ManagedBlob, not subclasses

Reply:
You're mistaken, `instanceof` also works with subclassing.